internal/mutators/namespace.go

```diff
@@ -52,19 +53,66 @@ func (m *Namespace) Handle(ctx context.Context, req admission.Request) admission
 // Currently, we only add `included-namespace` label to non-excluded namespaces
 // if the label is missing.
 func (m *Namespace) handle(log logr.Logger, ns *corev1.Namespace) {
-	// Early exit if the namespace is excluded.
-	if config.ExcludedNamespaces[ns.Name] {
-		return
-	}
 
-	// Add label if the namespace doesn't have it.
-	if _, hasLabel := ns.Labels[api.LabelIncludedNamespace]; !hasLabel {
-		if ns.Labels == nil {
-			ns.Labels = map[string]string{}
+	// if includedNamespaces is set, we dont need excluded as everyone that is not included is
+	// automatically excluded.
+	// if includedNamespaces is set, we need to set the label only to the included namespaces list
+
+	// if parameter included-namespaces is set, everything else is excluded
+	// so, we set the annotation only if ns exists in the map
+
+	//log.Info("we are in the mutator handle and this is the excluded map:", config.ExcludedNamespaces)
+
+	if config.IncludedNamespaces != nil {
+
+		// work with regulars
+		//log.Info("we are in the mutator handle, inside included process", config.IncludedNamespaces)
+
+		for nsRegexp, _ := range config.IncludedNamespaces {
+
+			//log.Info("this is the ns in the included map", nsRegexp)
+			if _, ok := config.ExcludedNamespaces[ns.Name]; ok {
+				//log.Info("it is already excluded, not setting labels", ns.Name)
+				return
+			}
+			match, _ := regexp.MatchString(nsRegexp, ns.Name)
+			if !match {
+				//log.Info("not matched , returned")
+				config.ExcludedNamespaces[ns.Name] = true
+				return
+			} else {
+				//log.Info("regex matched")
+				config.IncludedNamespaces[ns.Name] = true
+				if _, ok := config.IncludedNamespaces[ns.Name]; ok {
```

Comment:
when is this going to not enter the if condition? We just set it to true the line above